spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/reload/ConfigReloadUtil.java

```diff
@@ -108,14 +108,16 @@ static List<MapPropertySource> locateMapPropertySources(PropertySourceLocator pr
 
 		List<MapPropertySource> result = new ArrayList<>();
 		PropertySource<?> propertySource = propertySourceLocator.locate(environment);
-		if (propertySource instanceof MapPropertySource) {
-			result.add((MapPropertySource) propertySource);
+		if (propertySource instanceof MapPropertySource mapPropertySource) {
+			result.add(mapPropertySource);
 		}
 		else if (propertySource instanceof CompositePropertySource source) {
 
-			List<MapPropertySource> list = source.getPropertySources().stream()
-					.filter(p -> p instanceof MapPropertySource).map(x -> (MapPropertySource) x).toList();
-			result.addAll(list);
+			source.getPropertySources().forEach(x -> {
```

Comment:
this is so much easier to read, I am really sorry it did not occur to me when I was here in this code path before